draft-ietf-tls-esni.md

```diff
@@ -804,6 +804,17 @@ by attackers on the local network, which is a common case where ClientHello and
 encryption are desired. Moreover, as noted in the introduction, SNI encryption is
 less useful without encryption of DNS queries in transit via DoH or DPRIVE mechanisms.
 
+## Client Tracking
+
+A malicious client-facing server could distribute unique, per-client ECHOConfig
+structures as a way of tracking clients across subsequent connections. On-path
+adversaries which know about these unique keys could also track clients in this
+way by observing TLS connection attempts. This type of attack scales linearly
+with the desired number of target clients. Also, DNS caching behavior makes
+distribution of per-client ECHOConfig structures via HTTPSSVC RRs with high TTLs
+challenging. Moreover, given the existence of other tracking vectors, such as
+IPv6 addresses, this does not pose an immediate threat to clients.
```

Comment:
I don't think the v6 example is good. And none of these are especially strong, except perhaps the DNS one. 

For me, the question is how targeted the delivery could be, and whether the client can then do anything about that. For instance, a client might decide to flush its ECHO config cache if it moves and wants to be unlinkable.

Reply:
@martinthomson I'll remove the last sentence, not because of v6 (which I do think is a good example), but because it can probably bite us in the future when someone does turn it into a threat. :)

On the latter points, how about the following:

~~~
The cost of this type of attack scales linearly with the desired number of target clients. DNS caching behavior makes targeting individual users for extended periods of time, e.g., using per-client ECHOConfig structures delivered via HTTPSSVC RRs with high TTLs, challenging. Moreover, clients can flush any ECHOConfig state upon changing networks to mitigate potential linkability problems.
~~~